.github/actions/publish/upload_to_splunkbase.py

```diff
@@ -0,0 +1,162 @@
+"""
+Uploads a version of an app to Splunkbase
+"""
+
+import argparse
+import json
+import logging
+import os
+import sys
+import tarfile
+from packaging.version import parse
+
+import boto3
+
+from utils.update_version import find_app_json_name
+from utils.api.splunkbase import (
+    APACHE2_LICENSE_STRING,
+    APACHE2_LICENSE_URL,
+    SGT_LICENSE_STRING,
+    SGT_LICENSE_URL,
+    Splunkbase,
+)
+
+NEW_APP_WARNING_MESSAGE = (
+    "Successfully uploaded a NEW APP to Splunkbase. "
+    "Please notify the Splunkbase team. "
+    "See: http://go/new-soar-app-in-splunkbase for more."
+)
+
+RELEASE_QUEUE_URL = os.getenv("RELEASE_QUEUE_URL")
+SOAR_APPS_TOKEN = os.getenv("SOAR_APPS_TOKEN")
+SPLUNKBASE_USER = os.getenv("SPLUNKBASE_USER")
+SPLUNKBASE_PASSWORD = os.getenv("SPLUNKBASE_PASSWORD")
+RELEASE_QUEUE_REGION = 'us-west-2'
+
+
+
+def parse_args():
+    help_str = " ".join(line.strip() for line in __doc__.splitlines())
+    parser = argparse.ArgumentParser(description=help_str)
+    parser.add_argument("app_repo_name", help="Name of the app's GitHub repo.")
+    return parser.parse_args()
+
+
+def get_release_notes(tarball, version):
+    with tarfile.open(tarball, "r") as tar:
+        filename = f"release_notes/{version}.md"
+        names = tar.getnames()
+        notes_for_version = [n for n in names if filename in n]
+        if notes_for_version:
+            return tar.extractfile(notes_for_version[0]).read()
+    return None
```

Comment:
Type hints and some bike shedding:

```suggestion
def get_release_notes(tarball: str, version: str) -> Optional[str]:
    filename = f"release_notes/{version}.md"
    with tarfile.open(tarball, "r") as tar:
        for name in tar.getnames():
            if filename in name:
                return tar.extractfile(name).read()
    return None
```

Reply:
Will address in follow up